### PR TITLE
Replace Calypso FormCheckbox with CheckboxControl.

### PR DESCRIPTION
**Pull request by brezocordero**
## Description

Replace Calypso FormCheckbox with [CheckboxControl](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/checkbox-control/index.js).

### Related issue(s)

#2333 

### Steps to reproduce & screenshots/GIFs

* Go to Settings > Shipping > WooCommerce Shipping 
* Click on Choose a different credit card, verify that that the look and feel and behavior of the checkbox is unchanged.
* Verify that the checkbox for Email receipts looks and works as before.
CheckboxLabelPayment

* Click on Add Package, Service Package, verify that the look and feel and behavior of checkboxes is unchanged.
CheckboxesAddPackage

* Place an order with multiple packages. 
* Edit the order, create a label, on packaging, click on 'Move' on one of the packages, then add to a new package.
* Click on 'Add items', verify that that the look and feel and behavior of the checkbox is unchanged.
CheckboxAddToPackage

### Checklist

- [ ] unit tests

- [ ] `changelog.txt` entry added

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js [hunk 1]
```diff
@@ -320,22 +319,16 @@ class ShippingLabels extends Component {
...
 				</FormLabel>
-				<FormLabel>
-					<FormCheckbox
-						checked={ emailReceipts }
-						onChange={ onChange }
-						disabled={ ! canEditPayments && ! canEditSettings }
-					/>
-					<span className="label-settings__credit-card-description">
-						{ translate(
+					<CheckboxControl
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js]: Looks like the indentation is slightly off in here - weird that linter didn't catch it 😬

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js [hunk 2]
```diff
@@ -320,22 +319,16 @@ class ShippingLabels extends Component {
 		if ( ! isBoolean( emailReceipts ) ) {
 			return null;
 		}
-
+		const checkboxClasses = classNames( 'form-label', 'label-settings__credit-card-description' );
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js]: (minor) it doesn't look like we need the `classNames` utility - you can just do `className="form-label label-settings__credit-card-description"` inlined as a prop on the component.

File: client/extensions/woocommerce/woocommerce-services/views/packages/predefined-packages.js [hunk 3]
```diff
@@ -81,7 +84,7 @@ const PredefinedPackages = ( {
 					dimensionUnit={ dimensionUnit }
 					prefixActions
 				>
-					<FormCheckbox checked={ pckg.selected } onChange={ onToggle } />
+					<CheckboxControl className= { checkboxClasses } checked={ pckg.selected } onChange={ onToggle } />
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/views/packages/predefined-packages.js]: (minor)
```suggestion
					<CheckboxControl className={ checkboxClasses } checked={ pckg.selected } onChange={ onToggle } />
```

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/add-item.js [hunk 4]
```diff
@@ -55,15 +56,16 @@ const AddItemDialog = props => {
 			  } )
 			: item;
 
-		const onChange = event =>
-			props.setAddedItem( orderId, siteId, pckgId, itemIdx, event.target.checked );
+		const onChange = ( state ) => {
+			props.setAddedItem( orderId, siteId, pckgId, itemIdx, state );
+		}
+		const checkboxClasses = classNames( 'form-label', 'packages-step__dialog-package-option' );
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/add-item.js]: Also in this case, the `className` prop can be inlined on the component without the `classNames` utility

File: client/extensions/woocommerce/woocommerce-services/views/packages/style.scss [hunk 5]
```diff
@@ -208,6 +219,7 @@
 	width: 25%;
 	text-align: right;
 	padding-right: 16px;
+	margin: 0px;
```
Comment (frosso) [inline client/extensions/woocommerce/woocommerce-services/views/packages/style.scss]: (minor) the unit is not required when using `0`
```suggestion
	margin: 0;
```

**Review by frosso — COMMENTED** [review]
Changes look good 👍 

I noticed a little quirk (that is not present on `trunk`) with the checkboxes on the loading state of the settings page. Do you think it can be addressed?

Screen Shot 2021-07-09 at 12 29 41 PM

File: client/extensions/woocommerce/woocommerce-services/views/packages/style.scss (hunk 5 above)
Comment (brezocordero) [reply client/extensions/woocommerce/woocommerce-services/views/packages/style.scss]: Done!

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/add-item.js (hunk 4 above)
Comment (brezocordero) [reply client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/packages-step/add-item.js]: Done!

File: client/extensions/woocommerce/woocommerce-services/views/packages/predefined-packages.js (hunk 3 above)
Comment (brezocordero) [reply client/extensions/woocommerce/woocommerce-services/views/packages/predefined-packages.js]: Done!

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js (hunk 2 above)
Comment (brezocordero) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js]: Done!

File: client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js (hunk 1 above)
Comment (brezocordero) [reply client/extensions/woocommerce/woocommerce-services/views/label-settings/label-settings.js]: Done!

**Comment by brezocordero** [comment]
> Do you think it can be addressed?

Thanks for catching that, should be fixed now.

**Review by frosso — APPROVED** [review]
Peachy, thanks for the changes and for fixing that! :shipit:

**Pull request merged**